packages/cli/src/parallel-tests/parallel-tests.handler.ts

```diff
@@ -207,35 +123,26 @@ export const runAllTestsInParallel: (
             ? 0
             : resultsForTestCase.numberOfRetriesExecuted + 1;
 
+        resultsByTestId.set(id, {
+          currentResult: mergedResult,
+          numberOfRetriesExecuted,
+        });
+
         // Our work is done for this test case if the first result was a pass,
         // we've performed all the retries, or one of the retries already proved
         // the result as flakey
         const isFinalResult =
           mergedResult.result !== "fail" ||
           numberOfRetriesExecuted === maxRetriesOnFailure;
 
-        resultsByTestId.set(id, {
-          currentResult: mergedResult,
-          numberOfRetriesExecuted,
-        });
-
         if (isFinalResult) {
           // Cancel any replays that are still scheduled
-          queue = queue.filter(({ id }) => id !== id);
+          queue = queue.filter((otherReplay) => otherReplay.id !== id);
```

Comment:
Bug 3: Previously was comparing id to id! And so filtering everything out of the queue 🤦 .